Switch to chrome trace format for Tracer log viewing

Dr15Jones opened the pull request:
#### PR description:

- Switch to the json format supported by the built in Chrome trace viewer.
- Rework the non web json format.

#### PR validation:

Running some simple trace jobs through Chrome worked. Unit tests pass.

resolves https://github.com/cms-sw/framework-team/issues/1676
resolves https://github.com/cms-sw/framework-team/issues/1677
resolves https://github.com/cms-sw/framework-team/issues/1697
resolves https://github.com/cms-sw/framework-team/issues/1698
resolves https://github.com/cms-sw/framework-team/issues/1699

Dr15Jones committed:
Fix unit tests to handle new transitions

Dr15Jones committed:
Make json dump more human readable

Dr15Jones committed:
Remove unnecessary Counter

Turns out this was not actually being used in any meaningful way

Dr15Jones committed:
Use classes for json transition data

Dr15Jones committed:
Switch time to microseconds

Dr15Jones committed:
Add requesting information

Dr15Jones committed:
Added begin/end stream entries to tracer file

Dr15Jones committed:
Switch to chrome trace format for web output

Dr15Jones committed:
Remove old trace log web viewer

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones for master.

It involves the following packages:

- FWCore/Services (**core**)


@Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please review it and eventually sign? Thanks.
@fwyzard, @makortel, @wddgit this is something you requested to watch as well.
@ftenchini, @mandrenguyen, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
+1

**Size**: This PR adds an extra 44KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2cb7bd/49530/summary.html
**COMMIT**: ccb14f619e4676ec355d1a3a1e14a2efe8a90918
**CMSSW**: CMSSW_16_0_X_2025-11-18-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49408/49530/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-18-1100+2cb7bd/71836/validateJR/logRootQA-events.log) to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-18-1100+2cb7bd/71836/validateJR.html)
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 3896448
 - DQMHistoTests: [Total failures: 12](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-18-1100+2cb7bd/71836/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3896416
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 50 files compared)
 - Checked 218 log files, 188 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-18-1100+2cb7bd/71836/triggerResults): no differences found

Dr15Jones committed:
Capture all ES module IDs

The postESModuleRegistration appears to skip some ESModule reporting
(ones that are not used in the job?). Now gather the ids from the
constructor calls.

Dr15Jones committed:
Protect against a 'missing' member in module mode

Dr15Jones committed:
In module mode use activity name for displayed label

cmsbuild commented:
Pull request #49408 was updated. @Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please check and sign again.

Dr15Jones commented:
please test

Dr15Jones committed:
--web option now requires specifying name of output file

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49408/46913

cmsbuild commented:
Pull request #49408 was updated. @Dr15Jones, @makortel, @smuzaffar can you please check and sign again.

Dr15Jones committed:
Record module types in the tracer file

Dr15Jones committed:
Update edmTracerCompactLogViewer to handle module types

Dr15Jones commented:
No further changes are anticipated for this PR

makortel commented:
+core
